pkg/services/preference/prefapi/api.go

```diff
@@ -9,12 +9,13 @@ import (
 	"github.com/grafana/grafana/pkg/api/response"
 	"github.com/grafana/grafana/pkg/kinds/preferences"
 	"github.com/grafana/grafana/pkg/services/dashboards"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
 	pref "github.com/grafana/grafana/pkg/services/preference"
 )
 
 func UpdatePreferencesFor(ctx context.Context,
 	dashboardService dashboards.DashboardService, preferenceService pref.Service,
-	orgID, userID, teamId int64, dtoCmd *dtos.UpdatePrefsCmd) response.Response {
+	orgID, userID, teamId int64, dtoCmd *dtos.UpdatePrefsCmd, features featuremgmt.FeatureToggles) response.Response {
```

Comment:
I think we should reorder the arguments - the 'services' should come first.

```go
func UpdatePreferencesFor(ctx context.Context,
	dashboardService dashboards.DashboardService, preferenceService pref.Service, features featuremgmt.FeatureToggles,
	orgID, userID, teamId int64, dtoCmd *dtos.UpdatePrefsCmd) response.Response {
```

Call sites will need to be updated.